src/schema/schema_create.c

```diff
@@ -44,6 +44,58 @@ __wt_direct_io_size_check(
     return (0);
 }
 
+/*
+ * __check_imported_ts --
+ *     Check the aggregated timestamps for each checkpoint in a file that we've imported. We're not
+ *     allowed to import files with timestamps ahead of our oldest timestamp since a subsequent
```

Comment:
This comment is just saying that importing timestamps later than oldest shouldn't be allowed because:

### If an update is later than oldest, then it's within the queryable window of read timestamps and some of the results we return may be inconsistent with the old database.

In the old database with oldest=30, stable=50:
U6 (ts=40) -> U5 (ts=30)

In the new database with oldest=30, stable=50:
U6 (ts=40)

In this case, a read at timestamp 35 will return different results before and after import since we're not bringing across history. We don't have this issue if the updates must be behind oldest since we can't query behind it in the new database.

### If an update is later than stable, then rollback to stable after import will be destructive.

In the old database with oldest=30, stable=50:
U6 (ts=40) -> U5 (ts=30)

In the new database with oldest=30, stable=30:
U6 (ts=40)

A rollback to stable will remove this record which may not be expected.

In summary, we need to ensure that updates are earlier than stable in order to avoid the second issue. And we need to ensure that updates are earlier than oldest to avoid the first issue.

Reply:
Understood. Thanks for the explanation. we are not importing more than oldest timestamp, do we need to mention RTS here without mentioning a stable timestamp? Also, do we need to check whether the stable timestamp is less than the oldest timestamp?

Reply:
> Understood. Thanks for the explanation. we are not importing more than oldest timestamp, do we need to mention RTS here without mentioning a stable timestamp? Also, do we need to check whether the stable timestamp is less than the oldest timestamp?

When can that situation occur?
That doesn't make sense to me. I've always thought of stable as always being greater than the oldest.

Reply:
I think it is possible to set only the oldest timestamp but not stable, I feel WT still allows it. Maybe it is not a proper usage, so no need to worry.